solr/core/src/java/org/apache/solr/handler/admin/CollectionsHandler.java

```diff
@@ -476,6 +451,31 @@ private static void addStatusToResponse(NamedList<Object> results, RequestStatus
       SolrIdentifierValidator.validateAliasName(req.getParams().get(NAME));
       return req.getParams().required().getAll(null, NAME, "collections");
     }),
+    CREATEROUTEDALIAS_OP(CREATEROUTEDALIAS, (req, rsp, h) -> {
+      String alias = req.getParams().get(NAME);
+      SolrIdentifierValidator.validateAliasName(alias);
+      Map<String, Object> params = req.getParams().required()
+          .getAll(null, REQUIRED_ROUTING_PARAMS.toArray(new String[REQUIRED_ROUTING_PARAMS.size()]));
+      req.getParams().getAll(params, NONREQUIRED_ROUTING_PARAMS);
+      // subset the params to reuse the collection creation/parsing code
+      ModifiableSolrParams collectionParams = extractPrefixedParams("create-collection.", req.getParams());
+      if (collectionParams.get(NAME) != null) {
+        SolrException solrException = new SolrException(BAD_REQUEST, "routed aliases calculate names for their " +
+            "dependent collections, you cannot specify the name.");
+        log.error("Could not create routed alias",solrException);
+        throw solrException;
+      }
+      SolrParams v1Params = convertToV1WhenRequired(req, collectionParams);
+
+      // We need to add this temporary name just to pass validation.
```

Comment:
We need to insist that the collection config name is passed here.  It's very problematic to depend on automatic creation.  On a related note, when it's time to document TRA's, we may want to warn against data driven schema.

Reply:
The name on this line that I am TMP-ing isn't the collection config name, it's the name of the collection. We don't want them to pass it but the collection param parsing code I'm reusing will barf (rightly) if there's no name for a collection. We'll fix that up later when we calculate the name for the initial collection in the Cmd class.

Reply:
I wasn't clear; I don't have any issue with this part of the patch. I'm suggesting _adding_ a check that we require the configSet.  I needed to click on some line to insert my comment; maybe I should have put my review comment on the blank line.

Reply:
ah that's actually checked here: https://github.com/apache/lucene-solr/pull/304/files#diff-3fe6a8aeb14a57e63507fa17f8346771R207, but It could be moved to this class (or done both places)